storage/innobase/dict/dict0stats.cc

```diff
@@ -1763,32 +1846,36 @@ dict_stats_scan_page(
 		? page_rec_get_next_non_del_marked<true>
 		: page_rec_get_next_non_del_marked<false>;
 
-	const bool	should_count_external_pages = n_external_pages != NULL;
-
-	if (should_count_external_pages) {
-		*n_external_pages = 0;
+	if (leaf) {
+		leaf_stats->n_external_pages = 0;
```

Comment:
Shouldn’t this be already zero-initialized in the constructor? Better add
```c++
ut_ad(!leaf_stats->n_external_pages);
```
to be executed in each case (also `leaf=false`). I’d prefer a shorter name, such as `n_blobs`.